### ISSUE TITLE
WASAPI Loopback as Input Device

### ISSUE THREAD

**Issue by nomeat1337** [essej/sonobus#241]
Would it be possible to add the WASAPI loopback as an input device on windows ? 

Something like the implementation of audacity.

I was trying to send system audio over the network, but don't have stereo mix device. As of now, I can only send the audio from mic over the network.

**Comment by rob-fi** [comment]
I'm also looking for this feature. It would allow similar use as Audio Share (https://github.com/mkckr0/audio-share) as well as all the other bells and whistles that SonoBus provides.

(Actually it would be extra cool if you could send both the system audio loopback and a physical audio device at the same time, but at least WASAPI loopback devices being availabe in the device selection for now would be fine. Could always run a second instance of SonoBus for extra devices, I presume)

**Comment by vertexgamer** [comment]
Yeah i'm also looking at this feature, currently i'm forced to use windows stereo mix or voicemeeter to pass thru audio

**Comment by Raywonder** [comment]
The plugin app2clap found here can do that. https://app2clap.jantrid.net

**Comment by tecog** [comment]
I used VB-CABLE together with Sonobus to achieve the same functionality, but the latency is relatively high. I’m wondering if there’s a more efficient method.

**Comment by mthwJsmith** [comment]
Hi, got this working on #280 pull request. Working great for me and has the loopback devices for all your audio devices. Full WASAPI implementation. 

I did work on a branch that included application level loopback (e.g. select one application and it only loopsback that application), but this had sound issues that I didnt want to spend too long debugging, so have left it as a branch.

For my setup I send PC2's audio to a 'dummy' hardware speaker (the monitor speakers as its connected by hdmi) and set the hardware volume of that to 0. Then sonobus uses the loopback of that as the input, and this sends the audio to my PC 1. 

If you have no hardware audio output for whatever reason, you could use https://github.com/VirtualDrivers/Virtual-Audio-Driver as a "fake" speaker output
